media/js/base/stub-attribution.js

```diff
@@ -0,0 +1,240 @@
+     */
+    StubAttribution.getAttributionRate = function() {
+        var rate = $('html').attr('data-stub-attribution-rate');
+        return isNaN(rate) ? 0 : Math.min(Math.max(parseFloat(rate), 0), 1);
+    };
+
+    /**
+     * Returns true if both session cookies exist.
+     * @return {Boolean} data.
+     */
+    StubAttribution.hasSessionCookie = function() {
+        return Mozilla.Cookies.hasItem(StubAttribution.COOKIE_CODE_ID) && Mozilla.Cookies.hasItem(StubAttribution.COOKIE_SIGNATURE_ID);
+    };
+
+    /**
+     * Stores a session cookie with stub attribution data values.
+     * @param {Object} data - attribution_code, attribution_sig.
+     */
+    StubAttribution.setSessionCookie = function(data) {
+
+        if (!data.attribution_code || !data.attribution_sig) {
+            return;
+        }
+
+        Mozilla.Cookies.setItem(StubAttribution.COOKIE_CODE_ID, data.attribution_code);
+        Mozilla.Cookies.setItem(StubAttribution.COOKIE_SIGNATURE_ID, data.attribution_sig);
+    };
+
+    /**
+     * Gets stub attribution data from session cookie.
+     * @return {Object} - attribution_code, attribution_sig.
+     */
+    StubAttribution.getSessionCookie = function() {
+        return {
+            /* eslint-disable camelcase */
+            attribution_code: Mozilla.Cookies.getItem(StubAttribution.COOKIE_CODE_ID),
+            attribution_sig: Mozilla.Cookies.getItem(StubAttribution.COOKIE_SIGNATURE_ID)
+            /* eslint-enable camelcase */
+        };
+    };
+
+    /**
+     * Updates all download links on the page with additional query params for
+     * stub attribution.
+     * @param {Object} data - attribution_code, attribution_sig.
+     */
+    StubAttribution.updateBouncerLinks = function(data) {
+        /**
+         * If data is missing or the browser does not meet requirements for
+         * stub attribution, then do nothing.
+         */
+        if (!data.attribution_code || !data.attribution_sig || !StubAttribution.meetsRequirements()) {
+            return;
+        }
+
+        $('a[href*="https://download.mozilla.org/"]').each(function() {
+            this.href = Mozilla.StubAttribution.appendToDownloadURL(this.href, data);
+        });
+    };
+
+    /**
+     * Appends stub attribution data as URL parameters.
+     * Note: data is already URI encoded when returned via the service.
+     * @param {String url - URL to append data to.
+     * @param {Object} data - attribution_code, attribution_sig.
+     * @return {String} url + additional parameters.
+     */
+    StubAttribution.appendToDownloadURL = function(url, data) {
+
+        if (!data.attribution_code || !data.attribution_sig) {
+            return url;
+        }
+
+        // append stub attribution query params.
+        $.each(data, function(key, val) {
+            if (key === 'attribution_code' || key === 'attribution_sig') {
+                url += (url.indexOf('?') > -1 ? '&' : '?') + key + '=' + val;
```

Comment:
Yes I don't think it's strictly necessary here, but also it does no real harm leaving this here for completeness imo.